src/geometry/ArborX_GeometryTraits.hpp

```diff
@@ -111,21 +106,16 @@ void check_valid_geometry_traits(Geometry const &)
           GeometryTraits::dimension_v<Geometry> > 0,
       "GeometryTraits::dimension<Geometry>::value must be a positive integral");
 
-  static_assert(!std::is_same_v<typename tag<Geometry>::type, not_specialized>,
+  static_assert(!std::is_same_v<tag_t<Geometry>, not_specialized>,
                 "GeometryTraits::tag<Geometry> must define 'type' member type");
-  using Tag = typename tag<Geometry>::type;
-  static_assert(std::is_same<Tag, PointTag>{} || std::is_same<Tag, BoxTag>{} ||
-                    std::is_same<Tag, SphereTag>{} ||
-                    std::is_same<Tag, TriangleTag>{} ||
-                    std::is_same<Tag, KDOPTag>{},
+  static_assert(is_valid_geometry<Geometry>,
```

Comment:
`is_valid_geometry` includes rays.  Did you mean to add?
Now I see the message.  So it is a yes.
Why so?

Reply:
Because `Ray` is a valid geometry. However, `check_valid_geometry_traits` and the tag are a bit weird. Tags are useful within ArborX, but less so for user defined geometries. So, the type check should really be only called for the geometries we control.

Reply:
Was there a use case?

Reply:
No.